opa/tests/test.sh

```diff
@@ -1,36 +1,38 @@
 #!/bin/sh
+#/ Usage: test.sh <pkg_ident>
+#/
+#/ Example: test.sh core/opa/0.11.0/20190531065119
+#/
 
-TESTDIR="$(dirname "${0}")"
-PLANDIR="$(dirname "${TESTDIR}")"
-SKIPBUILD=${SKIPBUILD:-0}
+set -euo pipefail
+
+if [[ -z "${1:-}" ]]; then
+  grep '^#/' < "${0}" | cut -c4-
+  exit 1
+fi
 
+TESTDIR="$(dirname "${0}")"
 source "${TESTDIR}/helpers.bash"
 
-hab pkg install --binlink core/bats
+export TEST_PKG_IDENT="${1}"
+hab pkg install "${TEST_PKG_IDENT}"
 
 hab pkg install core/busybox-static
 hab pkg binlink core/busybox-static nc
-hab pkg install -b core/jq-static core/curl
+hab pkg install core/bats core/curl core/jq-static --binlink
 
-# Wait for supervisor to start
-echo "Waiting for supervisor to start"
-wait_listen tcp 9632 30
-
-source "${PLANDIR}/plan.sh"
-# Unload the service if its already loaded.
-hab svc unload "${HAB_ORIGIN}/${pkg_name}"
+hab sup term
+while hab sup status 2>/dev/null; do
+  echo "Waiting for supervisor to terminate"
+  sleep 1
+done
 
-if [ "${SKIPBUILD}" -eq 0 ]; then
-  set -e
-  pushd "${PLANDIR}" > /dev/null
-  build
-  popd > /dev/null
-  set +e
-fi
+hab sup run &
 
-source "${PLANDIR}/results/last_build.env"
-hab pkg install --binlink --force "${PLANDIR}/results/${pkg_artifact}"
-hab svc load "${pkg_ident}"
+echo "Waiting for supervisor to start (30s)"
+wait_listen tcp 9632 30
+hab svc unload "${TEST_PKG_IDENT}"
```

Comment:
On a fresh setup, I need to comment out this `unload` or I get this error:

```
Supervisor not started.
Waiting for supervisor to start (30s)
∵ Missing package for core/hab-launcher
» Installing core/hab-launcher
☁ Determining latest version of core/hab-launcher in the 'stable' channel
☛ Verifying core/hab-launcher/11300/20190605211433
→ Using core/bzip2/1.0.6/20190115011950
→ Using core/cacerts/2018.12.05/20190115014206
→ Using core/gcc-libs/8.2.0/20190115011926
→ Using core/glibc/2.27/20190115002733
→ Using core/libarchive/3.3.3/20190305214120
→ Using core/libsodium/1.0.16/20190116014025
→ Using core/linux-headers/4.17.12/20190115002705
→ Using core/openssl-fips/2.0.16/20190115014207
→ Using core/openssl/1.0.2r/20190305210149
→ Using core/xz/5.2.4/20190115013348
→ Using core/zlib/1.2.11/20190115003728
✓ Installed core/hab-launcher/11300/20190605211433
★ Install of core/hab-launcher/11300/20190605211433 complete with 1 new packages installed.
hab-sup(MR): core/hab-sup (core/hab-sup/0.82.0/20190605220053)
hab-sup(MR): Supervisor Member-ID 5ca4734261d34229a84b2d80de55b4ef
hab-sup(MR): Starting gossip-listener on 0.0.0.0:9638
hab-sup(MR): Starting ctl-gateway on 127.0.0.1:9632
hab-sup(MR): Starting http-gateway on 0.0.0.0:9631
✗✗✗
✗✗✗ [Err: 0] hab-sup(CMD)[components/sup/src/manager/commands.rs:265:22]: Service core/opa/0.11.0/20190611135403 not loaded
✗✗✗
```

In other PRs like https://github.com/habitat-sh/core-plans/pull/2600/files#diff-c33b12fb7ba336969cf19336928a31f4R22 the service unload is put at the end of the script and or'd with a true like

```
hab svc load "${TEST_PKG_IDENT}"

# Allow service start
WAIT_SECONDS=5
echo "Waiting ${WAIT_SECONDS} seconds for service to start..."
sleep "${WAIT_SECONDS}"

bats "$(dirname "${0}")/test.bats"

hab svc unload "${TEST_PKG_IDENT}" || true
```

The "or true" is to stop CI failing if the tests 100% pass but something goes wrong unloading the service.